topology/graph/indexer.go

```diff
@@ -204,7 +204,7 @@ func NewMetadataIndexer(g *Graph, listenerHandler ListenerHandler, m ElementMatc
 		Indexer: NewIndexer(g, listenerHandler, func(n *Node) (kv map[string]interface{}) {
 			if match := n.MatchMetadata(m); match {
 				kv = make(map[string]interface{})
-				if values, err := getFieldsAsArray(n, indexes); err == nil && len(values) > 0 {
+				if values, err := getFieldsAsArray(n, indexes); err == nil {
```

Comment:
Why ???

Reply:
@hunchback That's why before 0b81a07 we had a special case when len(indexes) == 0. Anyway, I'm against removing && len(values) > 0 as it simply makes no sense to have an empty list (values) in kv.